File: src/main/java/org/trp/shincolle/entity/base/SummonAiNumbers.java
package org.trp.shincolle.entity.base;

final class SummonAiNumbers {
    static final int LIFETIME_TICKS = 1200;
    static final int RETURN_STUCK_TICK_LIMIT = 120;
    static final int RETURN_FAILSAFE_TICKS = 20 * 30;
    static final int RETURN_TELEPORT_COOLDOWN_TICKS = 100;
    static final double RETURN_TELEPORT_DISTANCE_SQ = 256.0D;
    static final double RETURN_MOVE_SPEED = 1.2D;
    static final double RETURN_REACH_DISTANCE_SQ = 4.0D;
    static final int RETURN_MIN_MISSION_TICKS = 40;
    static final int RETURN_RECOVERY_POLL_INTERVAL = 20;
    static final double TELEPORT_VERTICAL_OFFSET = 0.75D;

    static final double ATTACK_MOVE_SPEED = 1.2D;
    static final int ATTACK_BEHAVIOR_PRIORITY = 1;
    static final int FOLLOW_BEHAVIOR_PRIORITY = 2;
    static final int ATTACK_DELAY_TICKS = 20;
    static final float ATTACK_LOOK_YAW = 30.0F;
    static final float ATTACK_LOOK_PITCH = 30.0F;
    static final double FOLLOW_CARRIER_DISTANCE_SQ = 64.0D;
    static final double FOLLOW_CARRIER_SPEED = 1.2D;
    static final int FOLLOW_RECALC_TICKS = 10;
    static final double RANDOM_STROLL_SPEED = 1.0D;
    static final float LOOK_AT_PLAYER_DISTANCE = 8.0F;
    static final int LOOK_AT_PLAYER_PRIORITY = 4;
    static final int RANDOM_LOOK_PRIORITY = 5;
    static final int RANDOM_STROLL_PRIORITY = 6;
    static final int RANDOM_LOOK_CHANCE = 50;
    static final int RANDOM_STROLL_CHANCE = 120;
    static final int RANDOM_STROLL_HORIZONTAL_RANGE = 10;
    static final int RANDOM_STROLL_VERTICAL_RANGE = 7;
    static final double INIT_SUMMON_OFFSET_RANGE = 3.0D;
    static final double INIT_SUMMON_OFFSET_CENTER = 1.5D;
    static final float DEFAULT_ATTACK_RANGE_SQ = 16.0F;
    static final double FOLLOW_RANGE_ATTR = 32.0D;
    static final float HEALTH_SCALE_FACTOR = 0.2F;
    static final float HEALTH_BASE = 10.0F;
    static final float SPEED_SCALE_FACTOR = 0.1F;
    static final float SPEED_BASE = 0.25F;
    static final float DAMAGE_SCALE_FACTOR = 0.5F;
    static final float DAMAGE_MIN = 2.0F;
    static final float ATTACK_DAMAGE_DEFAULT = 4.0F;
    static final float ATTACK_DAMAGE_CARRIER_FACTOR = 0.35F;
    static final int INITIAL_LIGHT_AMMO = 6;
    static final float ATTACK_SOUND_VOLUME = 1.0F;
    static final float ATTACK_SOUND_PITCH = 1.0F;
    static final int CRIT_PARTICLE_COUNT = 8;
    static final double CRIT_PARTICLE_OFFSET = 0.2D;
    static final double CRIT_PARTICLE_SPEED = 0.1D;

    private SummonAiNumbers() {
    }
}

File: src/main/java/org/trp/shincolle/entity/base/SummonAiNumbers.kt
package org.trp.shincolle.entity.base

internal object SummonAiNumbers {
    const val LIFETIME_TICKS: Int = 1200
    const val RETURN_STUCK_TICK_LIMIT: Int = 120
    val RETURN_FAILSAFE_TICKS: Int = 20 * 30
    const val RETURN_TELEPORT_COOLDOWN_TICKS: Int = 100
    const val RETURN_TELEPORT_DISTANCE_SQ: Double = 256.0
    const val RETURN_MOVE_SPEED: Double = 1.2
    const val RETURN_REACH_DISTANCE_SQ: Double = 4.0
    const val RETURN_MIN_MISSION_TICKS: Int = 40
    const val RETURN_RECOVERY_POLL_INTERVAL: Int = 20
    const val TELEPORT_VERTICAL_OFFSET: Double = 0.75

    const val ATTACK_MOVE_SPEED: Double = 1.2
    const val ATTACK_BEHAVIOR_PRIORITY: Int = 1
    const val FOLLOW_BEHAVIOR_PRIORITY: Int = 2
    const val ATTACK_DELAY_TICKS: Int = 20
    const val ATTACK_LOOK_YAW: Float = 30.0f
    const val ATTACK_LOOK_PITCH: Float = 30.0f
    const val FOLLOW_CARRIER_DISTANCE_SQ: Double = 64.0
    const val FOLLOW_CARRIER_SPEED: Double = 1.2
    const val FOLLOW_RECALC_TICKS: Int = 10
    const val RANDOM_STROLL_SPEED: Double = 1.0
    const val LOOK_AT_PLAYER_DISTANCE: Float = 8.0f
    const val LOOK_AT_PLAYER_PRIORITY: Int = 4
    const val RANDOM_LOOK_PRIORITY: Int = 5
    const val RANDOM_STROLL_PRIORITY: Int = 6
    const val RANDOM_LOOK_CHANCE: Int = 50
    const val RANDOM_STROLL_CHANCE: Int = 120
    const val RANDOM_STROLL_HORIZONTAL_RANGE: Int = 10
    const val RANDOM_STROLL_VERTICAL_RANGE: Int = 7
    const val INIT_SUMMON_OFFSET_RANGE: Double = 3.0
    const val INIT_SUMMON_OFFSET_CENTER: Double = 1.5
    const val DEFAULT_ATTACK_RANGE_SQ: Float = 16.0f
    const val FOLLOW_RANGE_ATTR: Double = 32.0
    const val HEALTH_SCALE_FACTOR: Float = 0.2f
    const val HEALTH_BASE: Float = 10.0f
    const val SPEED_SCALE_FACTOR: Float = 0.1f
    const val SPEED_BASE: Float = 0.25f
    const val DAMAGE_SCALE_FACTOR: Float = 0.5f
    const val DAMAGE_MIN: Float = 2.0f
    const val ATTACK_DAMAGE_DEFAULT: Float = 4.0f
    const val ATTACK_DAMAGE_CARRIER_FACTOR: Float = 0.35f
    const val INITIAL_LIGHT_AMMO: Int = 6
    const val ATTACK_SOUND_VOLUME: Float = 1.0f
    const val ATTACK_SOUND_PITCH: Float = 1.0f
    const val CRIT_PARTICLE_COUNT: Int = 8
    const val CRIT_PARTICLE_OFFSET: Double = 0.2
    const val CRIT_PARTICLE_SPEED: Double = 0.1
}
